uamqp/aio/_authentication_async.py

```diff
@@ -0,0 +1,76 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#-------------------------------------------------------------------------
+from functools import partial
+
+from ..authentication import (
+    _generate_sas_access_token,
+    SASTokenAuth,
+    JWTTokenAuth
+)
+from ..constants import AUTH_DEFAULT_EXPIRATION_SECONDS
+
+try:
+    from urlparse import urlparse
+    from urllib import quote_plus  # type: ignore
+except ImportError:
+    from urllib.parse import urlparse, quote_plus
+
+
+async def _generate_sas_token_async(auth_uri, sas_name, sas_key, expiry_in=AUTH_DEFAULT_EXPIRATION_SECONDS):
```

Comment:
We had to make this async?

Reply:
right, this method serves as the async `get_token` method for async auth classes, in which case `get_token` must be a coroutine